services/venice-controller/src/main/java/com/linkedin/venice/controller/kafka/consumer/AdminConsumptionTask.java

```diff
@@ -574,7 +580,46 @@ private void executeMessagesAndCollectResults() throws InterruptedException {
             int perStorePendingMessagesCount = storeQueue == null ? 0 : storeQueue.size();
             pendingAdminMessagesCount += perStorePendingMessagesCount;
             storesWithPendingAdminMessagesCount += perStorePendingMessagesCount > 0 ? 1 : 0;
-            if (e instanceof CancellationException) {
+
+            // Check if the cause is VeniceNoStoreException
+            Throwable cause = e.getCause();
+            if (cause instanceof VeniceNoStoreException) {
+              // Get the retry count for this store and version combination
+              Map<Long, Integer> retryCountMap =
+                  storeRetryCountMap.computeIfAbsent(storeName, s -> new ConcurrentHashMap<>());
+              AdminOperationWrapper nextOp = storeQueue != null ? storeQueue.peek() : null;
+              long offset = nextOp != null ? nextOp.getOffset() : UNASSIGNED_VALUE;
+              int currentRetryCount = retryCountMap.getOrDefault(offset, 0);
+
+              if (currentRetryCount >= MAX_RETRIES_FOR_NONEXISTENT_STORE) {
+                // We've reached the maximum retry limit for this store/message, so remove it from the queue
+                if (storeQueue != null && !storeQueue.isEmpty()) {
+                  AdminOperationWrapper removedOp = storeQueue.remove();
+                  LOGGER.info(
```

Comment:
Is the goal just to make delete store admin messages idempotent? If that's the case can we limit the scope to skip only delete store admin messages for `VeniceNoStoreException` (probably don't even need the retry)? If not then what are other valid cases or partial failure where skipping the admin message when encountering `VeniceNoStoreException` is the expected/desired behavior instead of raising a concern and signal the operators?

Reply:
can we check graveyard to confirm if a store is actually deleted and skip the message?

Reply:
@xunyin8 Actually no; the current deletion flow is sending an update-store command to disable read and write first, and followed by a delete-store command, so the stuck admin message is the update-store one.
@majisourav99 Good idea. Let me try.

Reply:
@xunyin8 Would you prefer limiting it to delete-store and update-store command for now?

Reply:
Yes, I see the update now and that's good.